jax/_src/scipy/linalg.py

```diff
@@ -1041,3 +1042,41 @@ def toeplitz(c: ArrayLike, r: Optional[ArrayLike] = None) -> Array:
       (nrows,), (1,), 'VALID', dimension_numbers=('NTC', 'IOT', 'NTC'),
       precision=lax.Precision.HIGHEST)[0]
   return jnp.flip(patches, axis=0)
+
+
+_NULLSPACE_DOC = """\
+Because the size of the output of ``null_space`` is data-dependent, the function is not
+typically compatible with JIT. The JAX version adds the optional ``size`` argument which
+must be specified statically for ``jax.scipy.linalg.null_space`` to be used within some of JAX's
+transformations.
+"""
+_NULLSPACE_EXTRA_PARAMS = """
+size : int, optional
+    If specified, ``size`` elements of the null space are returned. If there are
+    fewer null space elements than ``size`` indicates, the return value will be padded with constant
+    0 vectors that belong by nature to the null space.
+"""
+
+@_wraps(scipy.linalg.null_space, lax_description=_NULLSPACE_DOC, extra_params=_NULLSPACE_EXTRA_PARAMS)
+def null_space(A: ArrayLike, rcond: Optional[float] = None,
+               size: Optional[int] = None) -> Array:
+  u, s, vh = svd(A, full_matrices=True)
+  M, N = u.shape[0], vh.shape[1]
+  if rcond is None:
+    rcond = jnp.finfo(s.dtype).eps * max(M, N)
+  tol = jnp.amax(s) * rcond
+  num = jnp.sum(s > tol, dtype=int)
+  if size is None:
+    size = core.concrete_or_error(int, num,
+    "The size argument of null_space must be statically specified "
+    "to use null_space within jitted JAX transformations.")
+  else:
+    size = min(size, vh.shape[0])
+  Q = lax.dynamic_slice_in_dim(vh, size, vh.shape[0] - size, axis=0)
```

Comment:
Since `size` is never dynamic, we can use straightforward indexing here:
```
Q = vh[vh.shape[0] - size:]
```
(not `[-size:]` because `size` might be zero)

Reply:
right makes sense

Reply:
but does it make sense to allow `size==0`?